src/shared/containers/WorkSpaceList.tsx

```diff
@@ -0,0 +1,69 @@
+import * as React from 'react';
+import { useAsyncEffect } from 'use-async-effect';
+import { Resource } from '@bbp/nexus-sdk';
+import { useNexusContext } from '@bbp/react-nexus';
+import TabList from '../components/Tabs/TabList';
+import DashboardList from './DashboardList';
+
+interface WorkspaceListProps {
+  workSpaceIds: string[];
+  orgLabel: string;
+  projectLabel: string;
+}
+
+const WorkspaceList: React.FunctionComponent<WorkspaceListProps> = ({
+  workSpaceIds,
+  orgLabel,
+  projectLabel,
+}) => {
+  const [workSpaces, setWorkspaces] = React.useState<Resource[]>([]);
+  const [selectedWorkspace, setSelectedWorkspace] = React.useState<Resource>();
+  const nexus = useNexusContext();
+  const selectWorkspace = (id: string) => {
+    const w = workSpaces.find(w => w['@id'] === id);
+    setSelectedWorkspace(w);
+  };
+  useAsyncEffect(async () => {
+    const workSpaceList: Resource[] = [];
+    for (let i = 0; i < workSpaceIds.length; i +=1) {
+      const workspace = await nexus.httpGet({ path : workSpaceIds[i]}) as Resource;
```

Comment:
can you add a try/catch around that, just in case one of the workspace does not resolve...